test/pages/AuthorImagePage.ts

```diff
@@ -0,0 +1,94 @@
+import { Locator, Page } from "@playwright/test";
+import path from "path";
+export class AuthorImage {
+  page: Page;
+  profileIcon: Locator;
+  myProfileLocator: Locator;
+  listLinkLocator: Locator;
+  imageLocator: Locator;
+  addAuthorPhotoLocator: Locator;
+  chooseFileLocator: Locator;
+  uploadLocator: Locator;
+  finishedButtonForAdd: Locator;
+  finishedButtonForManage: Locator;
+  manageAuthorImagesLocator: Locator;
+  manageLocator: Locator;
+  uploadedImagesLocator;
+  trashCanLocator: Locator;
+  saveForTrashcanButton: Locator;
+
+  constructor({ page }: { page: Page }) {
+    this.page = page;
+    this.profileIcon = page.getByAltText("My account");
+    this.myProfileLocator = page.getByRole("link", { name: "My Profile" });
+    this.listLinkLocator = page.locator(".results");
+    this.imageLocator = page.locator(".bookcover img");
+    this.addAuthorPhotoLocator = page.locator(
+      `//span[@class="desktop-only"]//div[@class="manageCovers"]`
+    );
+    this.chooseFileLocator = page
+      .getByRole("tabpanel", { name: "Add" })
+      .locator("iframe")
+      .contentFrame()
+      .getByRole("button", { name: "Choose a JPG, GIF or PNG on" });
+    this.uploadLocator = page
+      .getByRole("tabpanel", { name: "Add" })
+      .locator("iframe")
+      .contentFrame()
+      .getByRole("button", { name: "Upload" });
+    this.finishedButtonForAdd = page
+      .getByRole("tabpanel", { name: "Add" })
+      .locator("iframe")
+      .contentFrame()
+      .getByRole("button", { name: "Finished" });
+    this.finishedButtonForManage = page
+      .getByRole("tabpanel", { name: "Manage" })
+      .locator("iframe")
+      .contentFrame()
+      .getByRole("button", { name: "Finished" });
+    this.manageAuthorImagesLocator = page.locator(
+      ".coverPop.dialog--open.cboxElement"
+    );
+    this.manageLocator = page.getByRole("link", { name: "MANAGE" });
+    this.uploadedImagesLocator = page
+      .getByRole("tabpanel", { name: "Manage" })
+      .locator("iframe")
+      .contentFrame()
+      .locator("img");
+    this.trashCanLocator = page
+      .getByRole("tabpanel", { name: "Manage" })
+      .locator("iframe")
+      .contentFrame()
+      .locator(".trash");
+    this.saveForTrashcanButton = page
+      .getByRole("tabpanel", { name: "Manage" })
+      .locator("iframe")
+      .contentFrame()
+      .getByRole("button", { name: "Save" });
+  }
+
+  async add() {
+    await this.profileIcon.click();
+    await this.myProfileLocator.click();
+    await this.listLinkLocator.click();
+    await this.imageLocator.click();
+    await this.addAuthorPhotoLocator.click();
+    await this.chooseFileLocator.setInputFiles(
+      path.join(__dirname, "photo.png")
```

Comment:
better to have this as a parameter to the function, so that different files can be tested.
also maybe place `photo.png` into a separate folder because its not really a page